internal/utils/net.go

```diff
@@ -34,6 +45,12 @@ func init() {
 		}
 		privateSubnets = append(privateSubnets, subnet)
 	}
+
+	var err error
+	Resolver, err = dnscache.New(defaultFreq, defaultLookupTimeout)
+	if err != nil {
+		panic(fmt.Sprintf("failed to setup DNS cache: %v", err))
+	}
 }
 
 func ResolveHostIP(ctx context.Context, hostIP string) (subnets []string, private bool, _ error) {
```

Comment:
Ditto, but since it's already here 🤷🏻

Reply:
Do you mean that ResolveHostIP should take a resolver as an argument?  ResolveHostIP is used in the autopilot as well.

Reply:
Anywhere that needs to resolve host IPs in a specific way should have an optional resolver as a field instead of relying on an `internal` package, but that's a bigger refactor than I think we want to put time into.